frappe/search/website_search.py

```diff
@@ -85,16 +87,23 @@ def parse_result(self, result):
 		)
 
 
-def slugs_with_web_view():
+def slugs_with_web_view(_items_to_index):
 	all_routes = []
 	filters = { "has_web_view": 1, "allow_guest_to_view": 1, "index_web_pages_for_search": 1}
-	fields = ["name", "is_published_field"]
+	fields = ["name", "is_published_field", 'website_search_field']
 	doctype_with_web_views = frappe.get_all("DocType", filters=filters, fields=fields)
 
 	for doctype in doctype_with_web_views:
 		if doctype.is_published_field:
-			routes = frappe.get_all(doctype.name, filters={doctype.is_published_field: 1}, fields="route")
-			all_routes += [route.route for route in routes]
+			docs = frappe.get_all(doctype.name, filters={doctype.is_published_field: 1}, fields=["route", doctype.website_search_field, 'title'])
+			if doctype.website_search_field:
+				for doc in docs:
+					content = frappe.utils.md_to_html(getattr(doc, doctype.website_search_field))
+					soup = BeautifulSoup(content, "html.parser")
+					text_content = soup.text if soup else ""
+					_items_to_index += [frappe._dict(title=doc.title, content=text_content, path=doc.route)]
```

Comment:
Size of this depends on number of pages exists in the system. We can use generator expression here to reduce high memory usage.